tests/extension/oneapi_current_device/current_device.cpp

```diff
@@ -0,0 +1,114 @@
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2025 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include <thread>
+
+namespace current_device::tests {
+
+TEST_CASE(
+    "Test for \"get_current_device\" get default device, call function and "
+    "compare devices ") {
+#ifndef SYCL_EXT_ONEAPI_CURRENT_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_current_device extension is not supported "
+      "by an implementation");
+#else
+  if (sycl::device::get_devices().size() < 1) {
+    SKIP("Test requires at least one device");
+  }
+  const sycl::device default_device{sycl::default_selector_v};
+  const auto current_device =
+      sycl::ext::oneapi::experimental::this_thread::get_current_device();
+  CHECK(
+      std::is_same_v<std::remove_cv_t<decltype(current_device)>, sycl::device>);
+  CHECK(default_device == current_device);
+#endif
+}
+
+TEST_CASE(
+    "Test for \"set_current_device and get_current_device\" check that "
+    "get_current_device function returns the device set by "
+    "set_current_device function") {
+#ifndef SYCL_EXT_ONEAPI_CURRENT_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_current_device extension is not supported by an "
+      "implementation");
+#else
+  for (const auto& device : sycl::device::get_devices()) {
+    sycl::ext::oneapi::experimental::this_thread::set_current_device(device);
+    const auto current_device =
+        sycl::ext::oneapi::experimental::this_thread::get_current_device();
+    CHECK(std::is_same_v<std::remove_cv_t<decltype(current_device)>,
+                         sycl::device>);
+    CHECK(device == current_device);
+  }
+#endif
+}
+
+void thread_callback(const sycl::device& device_to_set,
+                     sycl::device& current_device, bool& signal_thread1,
+                     bool& signal_thread2) {
+  sycl::ext::oneapi::experimental::this_thread::set_current_device(
+      device_to_set);
+  signal_thread1 = true;
+  while (!signal_thread2) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
+  }
+  current_device =
+      sycl::ext::oneapi::experimental::this_thread::get_current_device();
+}
+
+TEST_CASE(
+    "Test for calling \"set_current_device and get_current_device in "
+    "different "
+    "threads\", check that each thread has its own current device") {
```

Comment:
Same comment as somewhere above.  `"different "` is too small string and should be concatenated with the strings on the next lines (and clang-format applied after that, of course). 
```suggestion
    "Test for calling \"set_current_device and get_current_device in "
    "different threads\", check that each thread has its own current device") {
```